src/systemcmds/motor_ramp/motor_ramp.cpp

```diff
@@ -326,6 +375,32 @@ int motor_ramp_thread_main(int argc, char *argv[])
 		PX4_ERR("can't open %s", dev);
 	}
 
+	/* get the number of servo channels */
+	if (px4_ioctl(fd, PWM_SERVO_GET_COUNT, (unsigned long)&_servo_count) < 0) {
+			PX4_ERR("PWM_SERVO_GET_COUNT");
+			return 1;
```

Comment:
`fd` is not closed here.

Reply:
@bkueng it is used again to restore pwms after the test. Should I close it here then run the test and reopen it after the test is completed?

Reply:
If this ioctl fails, there is not much point in continuing and you can close the fd and return directly (and similar for the others) - at this point you only request information, you don't make any output changes yet.